### ISSUE TITLE
CompilationWithAnalyzersOptions.ReportSuppressedDiagnostics property has no effect on DiagnosticSuppressor

### ISSUE THREAD

**Issue by safu9** [dotnet/roslyn#52953]
When `CompilationWithAnalyzersOptions.ReportSuppressedDiagnostics` is set to true, 
CompilationWithAnalyzers doesn't return diagnostics suppressed by pragma or SuppressMessageAttribute,
but return diagnostics suppressed by DiagnosticSuppressors.

[The docs for ReportSuppressedDiagnostics](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.diagnostics.compilationwithanalyzersoptions.reportsuppresseddiagnostics?view=roslyn-dotnet-3.9.0) say this flag filters diagnostics with `IsSuppressed`.
Diagnostics suppressed by DiagnosticSuppressors also have `IsSuppressed`, so I think the docs or the implementation must be corrected.

Example code:
```csharp
var compilationWithAnalyzers = compilation.WithAnalyzers(analyzers, new CompilationWithAnalyzersOptions(
    options,
    onAnalyzerException: null,
    concurrentAnalysis: false,
    logAnalyzerExecutionTime: false,
    reportSuppressedDiagnostics: false));  // here

var diagnostics = await compilationWithAnalyzers
    .GetAnalyzerSemanticDiagnosticsAsync(semanticModel, filterSpan: null, cancellationToken);
```

Expected Behavior:
`diagnostics` don't include any diagnostics suppressed by DiagnosticSuppressors.

Actual Behavior:
`diagnostics` include diagnostics suppressed by DiagnosticSuppressors.

**Comment by jinujoseph** [comment]
Duplicate of #41584
